src/Quasar/Metastore.purs

```diff
@@ -0,0 +1,62 @@
+{-
+Copyright 2017 SlamData, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-}
+
+module Quasar.Metastore where
+
+import Prelude
+
+import Control.Alt ((<|>))
+import Data.Argonaut (Json, decodeJson, jsonEmptyObject, (.?), (:=), (~>))
+import Data.Either (Either)
+
+data Metastore p
+  = H2Metastore String
+  | PostgresMetastore { host ∷ String, port ∷ Int, database ∷ String | p }
+
+instance showMetastore ∷ Show (Metastore e) where
+  show = case _ of
+    H2Metastore s → "(H2Metastore " <> show s <> ")"
+    PostgresMetastore { host, port, database } →
+      "(PostgresMetastore { host: " <> show host <> ", port: " <> show port <> ", database: " <> show database  <> "})"
```

Comment:
Could derive generic and then show for this? I've been moving towards that in here, since show instances are the worst. :smile:

Reply:
Oh, I guess not because of the extensible row.